src/components/Table/GridTable.stories.tsx

```diff
@@ -1849,3 +1853,144 @@ export function Headers() {
     </div>
   );
 }
+
+/**
+ * Shows how drag & drop reordering can be implemented with GridTable drag events
+ */
+export function DraggableRows() {
+  const nameColumn: GridColumn<Row> = {
+    header: "Name",
+    data: ({ name }) => ({ content: <div>{name}</div>, sortValue: name }),
+  };
+
+  const actionColumn: GridColumn<Row> = {
+    header: "Action",
+    data: () => <div>Actions</div>,
+    clientSideSort: false,
+  };
+
+  let rowArray: GridDataRow<Row>[] = new Array(26).fill(0);
+  rowArray = rowArray.map((elem, idx) => ({
+    kind: "data",
+    id: "" + (idx + 1),
+    order: idx + 1,
+    data: { name: "" + (idx + 1), value: idx + 1 },
+    draggable: true,
+  }));
+
+  const [rows, setRows] = useState<GridDataRow<Row>[]>([simpleHeader, ...rowArray]);
+
+  // also works with as="table" and as="virtual"
+  return (
+    <GridTable
+      columns={[nameColumn, actionColumn]}
+      onRowDrop={(draggedRow, droppedRow, indexOffset) => {
+        const tempRows = [...rows];
+        // remove dragged row
+        // console.log("onRowDrop");
+        // console.log("rows", rows);
+        const draggedRowIndex = tempRows.findIndex((r) => r.id === draggedRow.id);
+        if (draggedRowIndex === -1) {
+          console.error("draggedRowIndex is -1");
+          console.log("draggedRow", draggedRow);
+          console.log("draggedRowIndex", draggedRowIndex);
+          return;
+        }
+        const reorderRow = tempRows.splice(draggedRowIndex, 1)[0];
+        // console.log("reorder row removed");
+
+        const droppedRowIndex = tempRows.findIndex((r) => r.id === droppedRow.id);
+
+        // console.log("ondrop: " + draggedRow.id + " -> " + droppedRow.id);
+
+        if (draggedRowIndex === -1 || droppedRowIndex === -1) {
+          console.error("draggedRowIndex or droppedRowIndex is -1");
+          console.log("draggedRow", draggedRow);
+          console.log("draggedRowIndex", draggedRowIndex);
+          console.log("droppedRow", droppedRow);
+          console.log("droppedRowIndex", droppedRowIndex);
+        } else {
+          // console.log("insert row at index " + droppedRowIndex);
+          // console.log("indexOffset", indexOffset);
+          // insert it at the dropped row index
+          setRows([...insertAtIndex(tempRows, reorderRow, droppedRowIndex + indexOffset)]);
+        }
+      }}
+      rows={[...rows]}
+    />
+  );
+}
+
+export const DraggableWithInputColumns = newStory(
```

Comment:
curious @bsholmes - For these last 2 columns, the dragged item does not remain in the position it is dropped. Is that expected? Am I doing something wrong like dropping things where they are not allowed to be?

Reply:
I neglecting to implement the drop event handler in those stories, should be there now